main.go

```diff
@@ -119,18 +128,65 @@ func main() {
 				},
 			},
 			{
-				Name:  "quality",
+				Name:  "quality-print",
 				Usage: "print quality checks of the data contract",
 				Flags: []cli.Flag{
 					fileNameFlag,
+					qualitySpecFileNameFlag,
+					&cli.StringFlag{
+						Name:  "quality-specification-path",
+						Value: "quality.specification",
+						Usage: "definition of a custom path to the quality specification in your data contract",
+					}},
+				Action: func(ctx *cli.Context) error {
+					pathToSpecification := strings.Split(ctx.String("quality-specification-path"), ".")
+					return PrintQuality(ctx.String("file"), ctx.String("quality-file"), pathToSpecification)
+				},
+			},
+			{
+				Name:  "quality-init",
```

Comment:
Can this be integrated into `test`/`quality-check`?

You can probably prepare the quality folder and copy the checks inside as a temporary file, which you can delete after the execution.

Reply:
If you know how to do that, do not hesitate.
The point at stake is to allow several/many quality checks, without having to specify/prepare the quality checks (_e.g._, with Soda configuration) every time: we prepare the quality checks once (as it may take a long time) and we check the quality of data many times.

Note that, in the current version of `sodaQuality.go`, the preparation/initialization step just creates the `quality` temporary directory, if needed, and writes a YAML Soda configuration file so as to use a local DuckDB database to store the data sets to be checked.
The users have then to create the local DuckDB database, which stores the data sets to be checked against the rules specified in the data contract. That local DuckDB database file may be copied from some other places. Such a procedure is documented in https://github.com/data-engineering-helpers/data-contracts/tree/main/datacontract.com#check-the-data-quality-of-a-table-thanks-to-a-data-contract
That step may be time-consuming. That is why I preferred not to include the initialization step in the `test` command.
On the other hand, since the initialization step is currently not destroying any previous work, if already existing, it would NOT be harmful (destructive of previous work, wasteful of time/resources) to systematically include the initialization step in the `test` command.
I let you decide what you prefer. Just let me know and I can include the initialization step in the Soda quality check (`test`) command.

Reply:
> Note that, in the current version of [sodaQuality.go](https://github.com/data-engineering-helpers/datacontract-cli/blob/main/sodaQuality.go), the preparation/initialization step just creates the quality/ temporary directory, if needed, and writes a YAML Soda configuration file so as to use a local DuckDB database to store the data sets to be checked.

That's what I saw, and why I'm asking. I'm actually not sure, if the CLI should be responsible for setting up the test environment. Since Soda supports several other data sources next to DuckDB, we would need to aim to support all of them. And probably the same for other tools. In my opinion, the datacontract CLI should expect a completely set up data quality tool and just run the configured checks on it - in this case, copy the checks and run them in Soda as you already implemented it. Please correct me, if you think I'm on the wrong path here.

For the sake of simplicity, I think it's best to integrate everything into `test` for now.

> If you know how to do that, do not hesitate.

I'm not sure what this applies to? [The other comment](https://github.com/datacontract/cli/pull/3#discussion_r1357876873)?

Reply:
[17f321f](https://github.com/datacontract/cli/pull/3/commits/17f321fa7249949b278b054123b3b8011ed9225e)
The initialization step has been embedded into the quality check part, and that holds true only for the Soda wrapper for now. As a matter of fact, other quality checkers may not need initialization steps.

Reply:
> I'm not sure what this applies to? https://github.com/datacontract/cli/pull/3#discussion_r1357876873?

I was referring to the following comment:

> Can this be integrated into `test / quality-check`?

Indeed, I thought you were meaning, with the slash (`/`) character, some sort of levels in the commands, as the slash (`/`) is usually the separator in file-system folder structures.
I now realize that the (`/`) character was here meaning "or" and/or _aka_ (also known as): `test` or/aka `quality-check`. So, it is all clear by now, thanks!

Reply:
> [...] I'm actually not sure, if the CLI should be responsible for setting up the test environment. Since Soda supports several other data sources next to DuckDB, we would need to aim to support all of them. And probably the same for other tools. In my opinion, the datacontract CLI should expect a completely set up data quality tool and just run the configured checks on it - in this case, copy the checks and run them in Soda as you already implemented it. Please correct me, if you think I'm on the wrong path here.

Yes, I (of course) fully agree with that: the `datacontract` CLI should support many tools and should be as much tool-agnostic as possible, (at least) when it comes to quality checks.

On the other hand, most of the quality checker tools need some sort of configuration in order to properly work. And since those tools are wrapped within/executed by the `datacontract` CLI, the `datacontract` CLI has to "be aware" of that local configuration. To be more specific:
* The SodaCL CLI needs to be passed a configuration file (with the `-c` parameter), namely `soda-conf.yml` here: `soda scan -d duckdb_local -c soda-conf.yml contracts/data-contract-flight-route-quality.yaml`.
* And the underlying database, namely `duckdb_local` here, must be provided with the `-d` parameter, and match the database parameter specified in the SodaCL configuration section.

The `datacontract` CLI has to "be aware" of those parameters/configuration elements, so as to pass them to the underlying quality checker tool (SodaCL here).

Reply:
> > I'm not sure what this applies to? https://github.com/datacontract/cli/pull/3#discussion_r1357876873?
> 
> I was referring to the following comment:
> 
> > Can this be integrated into `test / quality-check`?
> 
> Indeed, I thought you were meaning, with the slash (`/`) character, some sort of levels in the commands, as the slash (`/`) is usually the separator in file-system folder structures. I now realize that the (`/`) character was here meaning "or" and/or _aka_ (also known as): `test` or/aka `quality-check`. So, it is all clear by now, thanks!

Sorry for the confusion. We should drop `quality-check` and just go with `test`. The slash was meant as an "or" ;)

Reply:
> > [...] I'm actually not sure, if the CLI should be responsible for setting up the test environment. Since Soda supports several other data sources next to DuckDB, we would need to aim to support all of them. And probably the same for other tools. In my opinion, the datacontract CLI should expect a completely set up data quality tool and just run the configured checks on it - in this case, copy the checks and run them in Soda as you already implemented it. Please correct me, if you think I'm on the wrong path here.
> 
> Yes, I (of course) fully agree with that: the `datacontract` CLI should support many tools and should be as much tool-agnostic as possible, (at least) when it comes to quality checks.
> 
> On the other hand, most of the quality checker tools need some sort of configuration in order to properly work. And since those tools are wrapped within/executed by the `datacontract` CLI, the `datacontract` CLI has to "be aware" of that local configuration. To be more specific:
> 
>     * The SodaCL CLI needs to be passed a configuration file (with the `-c` parameter), namely `soda-conf.yml` here: `soda scan -d duckdb_local -c soda-conf.yml contracts/data-contract-flight-route-quality.yaml`.
> 
>     * And the underlying database, namely `duckdb_local` here, must be provided with the `-d` parameter, and match the database parameter specified in the SodaCL configuration section.
> 
> 
> The `datacontract` CLI has to "be aware" of those parameters/configuration elements, so as to pass them to the underlying quality checker tool (SodaCL here).
> 
> A way to accommodate that need is to have an optional `configuration` parameter in the `quality` section of the data contract specification. It would give something like:
> 
> ```yaml
> quality:
>   type: SodaCL
>   configuration: "$ref: quality/soda-conf.yaml"
>   specification: "$ref: contracts/data-contract-flight-route-quality.yaml"
> ```
> 
> In the initialization step (when there is one) of the quality checks, a check would be made on the configuration file; if the configuration file exists (and seems valid?), nothing more needs to be done during the initialization step; otherwise, the configuration file would be generated as it is currently done by the [`sodaQualityInit()` function](https://github.com/data-engineering-helpers/datacontract-cli/blob/main/sodaQuality.go#L23).
> 
> Would that work for you?

This configuration is specific to the test execution, right? 

If not, how about using the [server object](https://datacontract.com/#server-object)? 

If yes, what about using a generic flag called "test-arguments", or something similar? E.g.:

```
datacontract test --test-arguments="-d duckdb_local -c soda-conf.yml" contracts/data-contract-flight-route-quality.yaml
```

Reply:
Yes, the following would work:
```bash
datacontract test --test-arguments="-d duckdb_local -c quality/soda-conf.yml" contracts/data-contract-flight-route-quality.yaml
```

Reply:
> If not, how about using the [server object](https://datacontract.com/#server-object)?

The server seems rather related to where the data is stored (_e.g._, S3, GCS, ADLS). But it makes think that it could also address other parts of the data stack as well, such as ingestion, data processing, data servicing (visualization, API), monitoring/observability, access control, catalog, metadata, etc. Moreover, a given data product could source several data sets, each one stored in/serviced by different technologies.

Reply:
> > If not, how about using the [server object](https://datacontract.com/#server-object)?
> 
> By the way, the server seems rather related to where the data is stored (_e.g._, S3, GCS, ADLS). But it makes think that it could also address other parts of the data stack as well, such as ingestion, data processing, data servicing (visualization, API), monitoring/observability, access control, catalog, metadata, etc. Moreover, a given data product could source several data sets, each one stored in/serviced by different technologies.
> 

It's meant for the consumer to know where to get data. Similar to the Open API Server Object. https://learn.openapis.org/specification/servers.html

Reply:
> Yes, the following would work:
> ```bash
> datacontract test --test-arguments="-d duckdb_local -c quality/soda-conf.yml" contracts/data-contract-flight-route-quality.yaml
> ```
> 
> Will you add that command option/parameter?
> 

Then let's go with it for now. I think I would be best if you add it directly in your branch?

Reply:
Here we are: [e31d018](https://github.com/datacontract/cli/pull/3/commits/e31d018fb1b43b33349ac93e15f45b9029d5d0b9)

Example of the new command line for quality checks with options for SodaCL:
```bash
$ ./datacontract test --test-options "-d duckdb_local -c quality/soda-conf.yml" --file contracts/data-contract-flight-route.yaml --quality-file contracts/data-contract-flight-route-quality.yaml
```

Reply:
Well done :)

Reply:
And will you merge that pull request, then? :)

Reply:
Didn't realize that it's final yet, because the title still starts with WIP. I will take another look soon.

Reply:
Actually, there are plenty of things still to be done. But I would rather merge the pull request as is now (as it is working, even though in a somewhat limited way), while not closing it, so that I/we can continue to build upon it.

Reply:
Sure!
Don't they already work today? Do you have some specific test failure in mind? The scenario I tested worked, but I haven't added specific tests (as I'm not good enough in Go yet). We should of course add some more tests.

Reply:
Yes, tests would be great as well, especially integration tests in this case.

You can see what's failing in the checks section below:
https://github.com/datacontract/cli/actions/runs/6536887208/job/18055317649?pr=3

Reply:
The tests have been fixed. Thanks!